dart/example/monster_my_game.sample_generated.dart

```diff
@@ -0,0 +1,456 @@
+// automatically generated by the FlatBuffers compiler, do not modify
+// ignore_for_file: unused_import, non_constant_identifier_names
+
+library my_game.sample;
+
+import 'dart:typed_data' show Uint8List;
+import 'package:flat_buffers/flat_buffers.dart' as fb;
+
+
+class Color {
+  final int value;
+  const Color._(this.value);
+
+  factory Color.fromValue(int value) {
+    if (value == null) return null;
+    if (value < minValue || maxValue < value) {
+      throw new RangeError.range(value, minValue, maxValue);
+    }
+    return values[value - minValue];
+  }
+
+  static const int minValue = 0;
+  static const int maxValue = 2;
+  static bool containsValue(int value) => minValue <= value && value <= maxValue;
+
+  static const Color Red = const Color._(0);
+  static const Color Green = const Color._(1);
+  static const Color Blue = const Color._(2);
+  static get values => [Red,Green,Blue,];
+
+  static const fb.Reader<Color> reader = const _ColorReader();
+
+  @override
+  String toString() {
+    return 'Color{value: $value}';
+  }
+}
+
+class _ColorReader extends fb.Reader<Color> {
+  const _ColorReader();
+
+  @override
+  int get size => 1;
+
+  @override
+  Color read(fb.BufferContext bc, int offset) =>
+      new Color.fromValue(const fb.Int8Reader().read(bc, offset));
+}
+
+class EquipmentTypeId {
+  final int value;
+  const EquipmentTypeId._(this.value);
+
+  factory EquipmentTypeId.fromValue(int value) {
+    if (value == null) return null;
+    if (value < minValue || maxValue < value) {
+      throw new RangeError.range(value, minValue, maxValue);
+    }
+    return values[value - minValue];
+  }
+
+  static const int minValue = 0;
+  static const int maxValue = 1;
+  static bool containsValue(int value) => minValue <= value && value <= maxValue;
+
+  static const EquipmentTypeId NONE = const EquipmentTypeId._(0);
+  static const EquipmentTypeId Weapon = const EquipmentTypeId._(1);
+  static get values => [NONE,Weapon,];
+
+  static const fb.Reader<EquipmentTypeId> reader = const _EquipmentTypeIdReader();
+
+  @override
+  String toString() {
+    return 'EquipmentTypeId{value: $value}';
+  }
+}
+
+class _EquipmentTypeIdReader extends fb.Reader<EquipmentTypeId> {
+  const _EquipmentTypeIdReader();
+
+  @override
+  int get size => 1;
+
+  @override
+  EquipmentTypeId read(fb.BufferContext bc, int offset) =>
+      new EquipmentTypeId.fromValue(const fb.Uint8Reader().read(bc, offset));
+}
+
+/// Vec3
+abstract class Vec3 {
+  double get x;
+  double get y;
+  double get z;
+
+  factory Vec3(List<int> bytes) {
+    fb.BufferContext rootRef = new fb.BufferContext.fromBytes(bytes);
```

Comment:
This appears to be creating a `BufferContext` for each `Vec3`, or? what exactly is this function meant to do? Why are bytes stored in ints?

Reply:
This is if you want to create a `Vec3` object from a list of bytes (the `List<int>` - Dart does not have a `byte` type, int storage is dynamic per the spec, similar to JavaScript).  That factory constructor should not re-allocate the bytes passed in if the list is actually a `Uint8List` from the typed_data library - otherwise it will create a copy of the data into a `Uint8List` for the user.

Reply:
Still not getting it.. where do these bytes come from and what do they represent?

Reply:
If you wanted to create a `Vec3` from a raw array of bytes, this would do it.  But we're not creating `BufferContext`s for each `Vec3` in a monster - we're using the constructor on the `Impl` class, which takes the existing `BufferContext` and an `offset`

Reply:
In other words, internally, the factory constructor for `Vec3` is never used.  It would only be used by an external end user who wanted to have a flatbuffer of just `Vec3`s.

Reply:
I suppose I could drop these for structs, but in this implementation, it's possible to build a "flat buffer" with just Vec3s in it (and no root table).  I believe the C implementation supports that as well.  My preference would be to document this as non-standard behavior rather than remove the ability to do it if that works for you.

Reply:
That creates something that is not a FlatBuffer, and is not something that is supported by most implementations afaik. I'd remove it, as it could confuse people, and there really is no need for it.

Reply:
So should this only appear for the `root_type` or can it appear on any `table` and just not `struct`s?

Reply:
If this is going to be a way to decode whole buffers, then yes, allowing it on all tables is nice to have.